src/ray/object_manager/object_manager.cc

```diff
@@ -278,6 +278,7 @@ void ObjectManager::PullEstablishConnection(const ObjectID &object_id,
 
   if (conn != nullptr) {
     PullSendRequest(object_id, conn);
+    connection_pool_.ReleaseSender(ConnectionPool::ConnectionType::MESSAGE, conn);
   }
```

Comment:
Is it possible that ReleaseSender will happen before the actual write in the WriteMessageAsync? Due to the asynchronicity? Or is it fine?

Reply:
Yes, it will often happen before the write completes, but the write will get buffered inside of `conn`.